lib/http1.c

```diff
@@ -775,16 +828,32 @@ void finalostream_send(h2o_ostream_t *_self, h2o_req_t *req, h2o_iovec_t *inbufs
     }
 
     /* count bytes_sent if other ostreams haven't counted */
-    if (req->bytes_counted_by_ostream == 0) {
-        for (i = 0; i != inbufcnt; ++i) {
-            req->bytes_sent += inbufs[i].len;
-        }
+    size_t bytes_to_be_sent = 0;
+    for (i = 0; i != inbufcnt; ++i) {
+        bytes_to_be_sent += inbufs[i].len;
     }
+    req->bytes_sent += bytes_to_be_sent;
 
     if (!self->sent_headers) {
         conn->req.timestamps.response_start_at = h2o_gettimeofday(conn->super.ctx->loop);
         if (conn->req.send_server_timing_header)
             h2o_add_server_timing_header(&conn->req);
+
+        if (should_use_chunked_encoding(req)) {
+            self->chunked.enabled = 1;
```

Comment:
Don't we need to do either of:
* set this flag to zero if `should_use_chunked_encoding` returns false
* clear the flag in `init_request`

Otherwise, I believe that there would be a chance of server using chunked encoding even when it should not, while responding to a succeeding HTTP request of a persistent connection.